apps/project/graphql/filters.py

```diff
@@ -1,20 +1,47 @@
 import strawberry
 import strawberry_django
+from django.db import models
+from django.db.models.expressions import Value
+from django.db.models.functions import Concat
 
 from apps.project.models import Organization, Project, ProjectAsset
 
 
 @strawberry_django.filters.filter(Project, lookups=True)
 class ProjectFilter:
     id: strawberry.auto
-    name: strawberry.auto
+    topic: strawberry.auto
+    project_number: strawberry.auto
+    region: strawberry.auto
     project_type: strawberry.auto
     requesting_organization_id: strawberry.auto
     is_featured: strawberry.auto
     status: strawberry.auto
     team: strawberry.auto
     is_private: strawberry.auto
 
+    # NOTE: This might be slow for large datasets, Consider using vector searching in future
+    @strawberry_django.filter_field
+    def name(
+        self,
+        queryset: models.QuerySet[Project],
+        value: str,
+        prefix: str,
+    ) -> tuple[models.QuerySet[Project], models.Q]:
+        queryset = queryset.alias(
+            _name=Concat(
+                models.F(f"{prefix}topic"),
+                Value(" - "),
+                models.F(f"{prefix}region"),
+                Value(" ("),
+                models.F(f"{prefix}project_number"),
+                Value(") "),
+                models.F(f"{prefix}requesting_organization__name"),
+                output_field=models.CharField(),
+            ),
+        )
```

Comment:
Let's make this block a re-usable function. We can move this section to the project model and reuse this with both filtering and sorting.